common/namespace/handler.go

```diff
@@ -550,18 +560,28 @@ func (d *HandlerImpl) UpdateNamespace(
 	if configurationChanged && activeClusterChanged && isGlobalNamespace {
 		return nil, errCannotDoNamespaceFailoverAndUpdate
 	} else if configurationChanged || activeClusterChanged || needsNamespacePromotion {
-		// set the versions
-		if configurationChanged {
-			configVersion++
-		}
 		if (needsNamespacePromotion || activeClusterChanged) && isGlobalNamespace {
 			failoverVersion = d.clusterMetadata.GetNextFailoverVersion(
 				replicationConfig.ActiveClusterName,
 				failoverVersion,
 			)
 			failoverNotificationVersion = notificationVersion
 		}
+		// set the versions
+		if configurationChanged {
+			configVersion++
+		}
+
+		if configurationChanged || activeClusterChanged {
```

Comment:
if activeClusterChanged?

Reply:
Transcribing offline discussion: Left check in place for safety, but added comment.